test/integration/shoots/networking/shoot_networking_test.go

```diff
@@ -0,0 +1,175 @@
+
+func validateFlags() {
+	if !StringSet(*shootName) {
+		Fail("You should specify a shootName to test against")
+	}
+
+	if !StringSet(*kubeconfig) {
+		Fail("you need to specify the correct path for the kubeconfig")
+	}
+
+	if !FileExists(*kubeconfig) {
+		Fail("kubeconfig path does not exist")
+	}
+}
+
+var _ = Describe("Shoot network testing", func() {
+	var (
+		shootGardenerTest      *ShootGardenerTest
+		gardenerTestOperations *GardenerTestOperation
+		testLogger             *logrus.Logger
+
+		resourcesDir = filepath.Join("..", "..", "resources")
+
+		name      = "net-test"
+		namespace = "default"
+	)
+
+	CBeforeSuite(func(ctx context.Context) {
+		// validate flags
+		validateFlags()
+		testLogger = logger.AddWriter(logger.NewLogger(*logLevel), GinkgoWriter)
+
+		var err error
+		shootGardenerTest, err = NewShootGardenerTest(*kubeconfig, nil, testLogger)
+		Expect(err).NotTo(HaveOccurred())
+
+		shoot := &gardencorev1alpha1.Shoot{ObjectMeta: metav1.ObjectMeta{Namespace: *shootNamespace, Name: *shootName}}
+		gardenerTestOperations, err = NewGardenTestOperationWithShoot(ctx, shootGardenerTest.GardenClient, testLogger, shoot)
+		Expect(err).NotTo(HaveOccurred())
+	}, InitializationTimeout)
+
+	CAfterSuite(func(ctx context.Context) {
+		err := gardenerTestOperations.ShootClient.Client().Delete(ctx, &appsv1.DaemonSet{ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: namespace}})
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				Expect(err).To(HaveOccurred())
+			}
+		}
+	}, FinalizationTimeout)
+
+	CAfterEach(func(ctx context.Context) {
+		gardenerTestOperations.AfterEach(ctx)
+	}, DumpStateTimeout)
+
+	CIt("should reach all webservers on all nodes", func(ctx context.Context) {
+		templateFilepath := filepath.Join(resourcesDir, "templates", nginxTemplateName)
+		nettestTmpl := template.Must(template.ParseFiles(templateFilepath))
+
+		By("Deploy the net test daemon set")
+		var writer bytes.Buffer
+		err := nettestTmpl.Execute(&writer, map[string]string{
+			"name":      name,
+			"namespace": namespace,
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		manifestReader := kubernetes.NewManifestReader(writer.Bytes())
+		err = gardenerTestOperations.ShootClient.Applier().ApplyManifest(ctx, manifestReader, kubernetes.DefaultApplierOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = gardenerTestOperations.WaitUntilDaemonSetIsRunning(ctx, name, namespace, gardenerTestOperations.ShootClient)
+		Expect(err).NotTo(HaveOccurred())
+
+		pods := &corev1.PodList{}
+		err = gardenerTestOperations.ShootClient.Client().List(ctx, pods, client.MatchingLabels{"app": "net-nginx"})
+		Expect(err).NotTo(HaveOccurred())
+
+		// check if all webservers can be reached from all nodes
+		By("test connectivity to webservers")
+		shootRESTConfig := gardenerTestOperations.ShootClient.RESTConfig()
+		var res error
+		for _, from := range pods.Items {
+			for _, to := range pods.Items {
+				By(fmt.Sprintf("Testing %s to %s", from.GetName(), to.GetName()))
```

Comment:
maybe we should also skip pinging to self:

```go
if from.Name == to.Name {
	continue
}
```

Reply:
I first wanted to do that but if I remember correctly we had some issues where only communication between pods on the very same node was not working. But all other communication.
Therefore I left it in to also cover this special case.
Do you remember that case?

Reply:
sure, but this is a Daemonset, you need two pods scheduled on the same node to do this test, correct?

In anycase, lets keep it for now, and add more tests to cover pod-to-pod later.

Reply:
yes your're right we're testing the connection to the same pod but as we use the pod IP wouldn't the connection like like this:  curl-container -> proxy -> pod -> nginx-container ?
So we would go through the pod network..

Reply:
it would not exit the node, it would just connect without the proxy hub, it is just a different port. 

In anycase, let's keep it for now, it might also be a good reflection check, in case all the other connections to other nodes are not working, we can still say at least that the application is running if it works from the same node.